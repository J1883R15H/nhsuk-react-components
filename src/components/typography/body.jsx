import React from 'react';
import { oneOf, node, string } from 'prop-types';
import stylePropType from 'react-style-proptype';

const Body = ({ size, children, className, style }) => {
  return (
    <p
      className={`nhsuk-body${
        size && size !== 'm' ? `-${size}` : ''
      } ${className}`}
      style={style}
    >
      {children}
    </p>
  );
};
Body.propTypes = {
  size: oneOf(['s', 'm', 'l']),
  children: node.isRequired,
  style: stylePropType,
  className: string
};

Body.defaultProps = {
  size: 'm',
  style: {},
  className: ''
};

export default Body;
