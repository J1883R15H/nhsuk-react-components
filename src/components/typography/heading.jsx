import React from 'react';
import { oneOf, node, string } from 'prop-types';
import classNames from 'classnames';

const Heading = ({ size, children, className, ...rest }) => {
  switch (size) {
    case 'xl':
      return (
        <h1 className={classNames('nhsuk-heading-xl', className)} {...rest}>
          {children}
        </h1>
      );
    case 'l':
      return (
        <h2 className={classNames('nhsuk-heading-l', className)} {...rest}>
          {children}
        </h2>
      );
    case 'm':
      return (
        <h3 className={classNames('nhsuk-heading-m', className)} {...rest}>
          {children}
        </h3>
      );
    case 's':
      return (
        <h4 className={classNames('nhsuk-heading-s', className)} {...rest}>
          {children}
        </h4>
      );
    case 'xs':
      return (
        <h5 className={classNames('nhsuk-heading-xs', className)} {...rest}>
          {children}
        </h5>
      );
    default:
      return null;
  }
};

Heading.propTypes = {
  size: oneOf(['xl', 'l', 'm', 's', 'xs']),
  children: node.isRequired,
  className: string
};

Heading.defaultProps = {
  size: 'xl',
  className: ''
};

export default Heading;
