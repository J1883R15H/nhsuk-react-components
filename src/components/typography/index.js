import Body from './body';
import Heading from './heading';
import List from './list';

export { Body, Heading, List };
