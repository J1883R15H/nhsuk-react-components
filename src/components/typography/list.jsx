import React from 'react';
import { oneOf, node, string } from 'prop-types';
import stylePropType from 'react-style-proptype';

const List = ({ type, children, className, style }) => {
  if (type === 'number') {
    return (
      <ol
        className={`nhsuk-list nhsuk-list--number ${className}`}
        style={style}
      >
        {children}
      </ol>
    );
  }
  return (
    <ul className={`nhsuk-list nhsuk-list--bullet ${className}`} style={style}>
      {children}
    </ul>
  );
};

List.propTypes = {
  type: oneOf(['bullet', 'number']),
  children: node.isRequired,
  className: string,
  style: stylePropType
};

List.defaultProps = {
  type: 'bullet',
  className: '',
  style: {}
};
