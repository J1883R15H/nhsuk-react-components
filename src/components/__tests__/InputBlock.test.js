import React from 'react';
import { shallow, mount } from 'enzyme';
import InputBlock from '../form/components/input-block';
import Hint from '../hint';
import Label from '../label';
import ErrorMessage from '../error-message';
import { Heading } from '../typography';

describe('InputBlock', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<InputBlock>Child</InputBlock>);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('has valid defaultProps', () => {
    expect(InputBlock.defaultProps.valueCallback()).toBe(undefined);
    expect(InputBlock.defaultProps.registerInitialValue()).toBe(undefined);
  });
  it('responds to change', () => {
    const valueCallback = jest.fn();
    const wrapper = mount(
      <InputBlock valueCallback={valueCallback}>
        <InputBlock.Input name="thing" />
      </InputBlock>
    );
    wrapper
      .find('input')
      .simulate('change', { target: { value: 'InputValue' } });
    expect(valueCallback).toHaveBeenCalled();
    expect(valueCallback.mock.calls[0][0]).toBe('thing');
    expect(valueCallback.mock.calls[0][1]).toBe('InputValue');
    wrapper.unmount();
  });
  it('renders heading', () => {
    const wrapper = shallow(<InputBlock heading="Title" headingSize="xl" />);
    expect(
      wrapper.containsMatchingElement(<Heading size="xl">Title</Heading>)
    ).toBeTruthy();
    wrapper.unmount();
  });
  describe('Input', () => {
    it('matches snapshot', () => {
      const wrapper = shallow(<InputBlock.Input name="thing" />);
      expect(wrapper).toMatchSnapshot();
      wrapper.unmount();
    });
    it('renders label, hint and error', () => {
      const wrapper = shallow(
        <InputBlock.Input
          name="thing"
          label="title"
          hint="hint"
          error="error"
        />
      );
      expect(
        wrapper.containsMatchingElement(<Label>title</Label>)
      ).toBeTruthy();
      expect(wrapper.containsMatchingElement(<Hint>hint</Hint>)).toBeTruthy();
      expect(
        wrapper.containsMatchingElement(<ErrorMessage>error</ErrorMessage>)
      ).toBeTruthy();
      wrapper.unmount();
    });
    it('changes width', () => {
      const wrapper = mount(<InputBlock.Input name="thing" width={12} />);
      expect(wrapper.find('input').props().className).toBe(
        'nhsuk-input nhsuk-input--block nhsuk-input--width-12'
      );
      wrapper.unmount();
    });
    it('has valid defaultProps', () => {
      expect(InputBlock.Input.defaultProps._handleInput()).toBe(undefined);
      expect(InputBlock.Input.defaultProps.registerInitialValue()).toBe(
        undefined
      );
    });
  });
});
