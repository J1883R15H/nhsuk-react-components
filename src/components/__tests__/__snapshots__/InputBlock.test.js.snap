// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputBlock Input matches snapshot 1`] = `
<Fragment>
  <input
    aria-describedby="thing-hint"
    autoComplete=""
    className="nhsuk-input nhsuk-input--block"
    id="thing"
    name="thing-hint"
    onChange={[Function]}
    placeholder=""
    style={Object {}}
    type="text"
  />
</Fragment>
`;

exports[`InputBlock matches snapshot 1`] = `
<div
  className=""
  style={Object {}}
>
  Child
</div>
`;
